applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1752 @@
+         * integration points related to different integration method
+         * implemented in geometry.
+         */
+        typedef typename BaseType::IntegrationPointsContainerType IntegrationPointsContainerType;
+
+        /**
+         * A third order tensor used as shape functions' values
+         * container.
+         */
+        typedef typename BaseType::ShapeFunctionsValuesContainerType ShapeFunctionsValuesContainerType;
+
+        /**
+         * A fourth order tensor used as shape functions' local
+         * gradients container in geometry.
+         */
+        typedef typename BaseType::ShapeFunctionsLocalGradientsContainerType ShapeFunctionsLocalGradientsContainerType;
+
+        /**
+         * A third order tensor to hold jacobian matrices evaluated at
+         * integration points. Jacobian and InverseOfJacobian functions
+         * return this type as their result.
+         */
+        typedef typename BaseType::JacobiansType JacobiansType;
+
+        /**
+         * A third order tensor to hold shape functions' local
+         * gradients. ShapefunctionsLocalGradients function return this
+         * type as its result.
+         */
+        typedef typename BaseType::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
+
+        /**
+         * A third order tensor to hold shape functions' local second derivatives.
+         * ShapefunctionsLocalGradients function return this
+         * type as its result.
+         */
+        typedef typename BaseType::ShapeFunctionsSecondDerivativesType
+            ShapeFunctionsSecondDerivativesType;
+
+        /**
+        * A third order tensor to hold shape functions' local third derivatives.
+        * ShapefunctionsLocalGradients function return this
+        * type as its result.
+        */
+        typedef typename BaseType::ShapeFunctionsThirdDerivativesType
+            ShapeFunctionsThirdDerivativesType;
+
+        /**
+         * Type of the normal vector used for normal to edges in geometry.
+         */
+        typedef typename BaseType::NormalType NormalType;
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+        
+        // ========================================================================================
+        // ========================================================================================
+        Triangle2D15(const PointType& Point01, const PointType& Point02, const PointType& Point03,
+            const PointType& Point04, const PointType& Point05, const PointType& Point06,
+            const PointType& Point07, const PointType& Point08, const PointType& Point09,
+            const PointType& Point10, const PointType& Point11, const PointType& Point12,
+            const PointType& Point13, const PointType& Point14, const PointType& Point15)
+            : BaseType(PointsArrayType(), &msGeometryData)
+        {
+            this->Points().push_back(typename PointType::Pointer(new PointType(Point01)));
```

Comment:
can't we create an aux variable called `auto& r_points = this->Points()` and then `r_points .push_back` it avoids calling all the time the this->Points()

Reply:
Done